sklearn/datasets/_california_housing.py

```diff
@@ -98,6 +106,10 @@ def fetch_california_housing(data_home=None, download_if_missing=True,
 
         .. versionadded:: 0.20
 
+    frame : pandas DataFrame
+        Only present when `as_frame=True`. DataFrame with ``data`` and
+        ``target``.
+
```

Comment:
Pleased add the directive:

```
.. versionadded:: 0.23
```

Reply:
Is this the right way to do it?
```python
def _pandas_is_missing():
    try:
        import pandas
    except ImportError:
        raise SkipTest('fetch_california_housing with as_frame=True'
                       ' requires pandas')


@pytest.mark.skipif(
    _pandas_is_missing(),
    reason='Import pandas library to run this test'
)
```

Reply:
In https://github.com/scikit-learn/scikit-learn/pull/15950#discussion_r360978781 I mentioned to add a new test to check that the warning is raised without skipping the test when pandas is missing. Something like the following (untested):

```python
@pytest.mark.skipif(
    _is_california_housing_dataset_not_available(),
    reason='Download California Housing dataset to run this test'
)
def test_pandas_dependency_message()
    try:
        import pandas
        pytest.skip("This test requires pandas to be not installed")
    except ImportError:
        # Check that pandas is imported lazily and that an informative error message is
        # raised when pandas is missing:
        expected_msg = "fetch_california_housing with as_frame=True requires pandas"
        with pytest.raises(ImportError, match=expected_msg):
            fetch_california_housing(as_frame=True)
```